File: src/Samples/Scheduler.ASP.NET/SaveRecurring.ashx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using DHTMLX.Common;
using DHTMLX.Helpers;
using System.Globalization;

namespace SchedulerNetAsp
{
    /// <summary>
    /// </summary>
    public class SaveRecurring : IHttpHandler
    {


        protected bool deleteRelated(DataAction action, Recurring changedEvent, SchedulerDataContext context)
        {
            bool finished = false;
            if ((action.Type == DataActionTypes.Delete || action.Type == DataActionTypes.Update) && !string.IsNullOrEmpty(changedEvent.rec_type))
            {
                context.Recurrings.DeleteAllOnSubmit(from ev in context.Recurrings where ev.event_pid == changedEvent.id select ev);
            }
            if (action.Type == DataActionTypes.Delete && changedEvent.event_pid != 0)
            {
                Recurring changed = (from ev in context.Recurrings where ev.id == action.TargetId select ev).Single();
                changed.rec_type = "none";
                finished = true;
            }
            return finished;
        }

        protected DataAction insertRelated(DataAction action, Recurring changedEvent, SchedulerDataContext context)
        {
            if (action.Type == DataActionTypes.Insert && changedEvent.rec_type == "none")
            {//insert_related
                 action.Type = DataActionTypes.Delete;
            }
            return action;
        }
        public void ProcessRequest(HttpContext context)
        {
            var action = new DataAction(context.Request.Form);
            var data = new SchedulerDataContext();

            try
            {

                var changedEvent = (Recurring)DHXEventsHelper.Bind(typeof(Recurring), context.Request.Form);//create model object from the request fields

                bool isFinished = deleteRelated(action, changedEvent, data);
                if (!isFinished)
                {
                    switch (action.Type)
                    {
                        case DataActionTypes.Insert: // define here your Insert logic
                            data.Recurrings.InsertOnSubmit(changedEvent);
                            
                            break;
                        case DataActionTypes.Delete: // define here your Delete logic
                            changedEvent = data.Recurrings.SingleOrDefault(ev => ev.id == action.SourceId);
                            data.Recurrings.DeleteOnSubmit(changedEvent);
                            break;
                        default:// "update" // define here your Update logic
                            var updated = data.Recurrings.SingleOrDefault(ev => ev.id == action.SourceId);
                            DHXEventsHelper.Update(updated, changedEvent, new List<string>() { "id" });
                            

                            break;
                    }
                }
                data.SubmitChanges();
                action.TargetId = changedEvent.id;
                action = insertRelated(action, changedEvent, data);
            }
            catch
            {
                action.Type = DataActionTypes.Error;
            }
            context.Response.ContentType = "text/xml";
            context.Response.Write(new AjaxSaveResponse(action).ToString());
        }

        public bool IsReusable
        {
            get
            {
                return false;
            }
        }
    }
}

File: src/Samples/Scheduler.ASP.NET.VB/SaveRecurring.ashx.vb
Imports System.Web
Imports System.Web.Services
Imports DHTMLX.Common

Public Class SaveRecurring
    Implements System.Web.IHttpHandler


    Protected Function deleteRelated(ByVal action As DataAction, ByVal changedEvent As Recurring, ByVal context As SchedulerDataContext) As Boolean
        Dim finished As Boolean = False
        If (action.Type = DataActionTypes.Delete OrElse action.Type = DataActionTypes.Update) AndAlso Not String.IsNullOrEmpty(changedEvent.rec_type) Then
            context.Recurrings.DeleteAllOnSubmit(From ev In context.Recurrings Where ev.event_pid = changedEvent.id)
        End If
        If action.Type = DataActionTypes.Delete AndAlso changedEvent.event_pid <> 0 Then
            Dim changed As Recurring = (From ev In context.Recurrings Where ev.id = action.TargetId).[Single]()
            changed.rec_type = "none"
            finished = True
        End If
        Return finished
    End Function

    Protected Function insertRelated(ByVal action As DataAction, ByVal changedEvent As Recurring, ByVal context As SchedulerDataContext) As DataAction
        If action.Type = DataActionTypes.Insert AndAlso changedEvent.rec_type = "none" Then
            'insert_related
            action.Type = DataActionTypes.Delete
        End If
        Return action
    End Function

    Sub ProcessRequest(ByVal context As HttpContext) Implements IHttpHandler.ProcessRequest
        context.Response.ContentType = "text/xml"
        Dim action = New DataAction(context.Request.Form)
        Dim data = New SchedulerDataContext()

        Try
            Dim changedEvent = DirectCast(DHXEventsHelper.Bind(GetType(Recurring), context.Request.Form), Recurring)
            Dim isFinished As Boolean = deleteRelated(action, changedEvent, data)
            If Not isFinished Then
                Select Case action.Type
                    Case DataActionTypes.Insert
                        ' define here your Insert logic
                        data.Recurrings.InsertOnSubmit(changedEvent)

                        Exit Select
                    Case DataActionTypes.Delete
                        ' define here your Delete logic
                        changedEvent = data.Recurrings.SingleOrDefault(Function(ev) ev.id = action.SourceId)
                        data.Recurrings.DeleteOnSubmit(changedEvent)
                        Exit Select
                    Case Else
                        ' "update" // define here your Update logic
                        Dim updated = data.Recurrings.SingleOrDefault(Function(ev) ev.id = action.SourceId)
                        DHXEventsHelper.Update(updated, changedEvent, New List(Of String)() From {"id"})


                        Exit Select
                End Select
            End If
            data.SubmitChanges()
            action.TargetId = changedEvent.id
            action = insertRelated(action, changedEvent, data)
        Catch a As Exception
            action.Type = DataActionTypes.[Error]
        End Try

        context.Response.Write(New AjaxSaveResponse(action).ToString())
    End Sub

    ReadOnly Property IsReusable() As Boolean Implements IHttpHandler.IsReusable
        Get
            Return False
        End Get
    End Property
End Class

